homeassistant/components/mochad.py

```diff
@@ -39,19 +42,33 @@ def setup(hass, config):
     conf = config[DOMAIN]
     host = conf.get(CONF_HOST)
     port = conf.get(CONF_PORT)
+    if setup_component(hass, MQTT_COMPONENT, DOMAIN_MQTT):
+        conf_mqtt = config[DOMAIN_MQTT]
```

Comment:
I'm not sure we can or should depend on the mqtt component config. The mqtt component can now be set up via config entry, so the config section is not required.

We could still use the mqtt component as transport if we just use its publish component function. Then we don't need to connect again to the broker so we don't need the mqtt config details. But the mochad mqtt library would have to support that though.

Reply:
Good point. I finally managed to find a way to reuse mqtt client in hass. Refactored to use the option of passing mqtt client as argument to pymochad_mqtt. Instead of instantiating a new mqtt client from the config. Cleaning out all the mqtt config references.